src/tools/gophish_export.py

```diff
@@ -253,6 +254,74 @@ def get_application(rawEvent):
     return application
 
 
+def find_unique_target_clicks_count(clicks):
+    """Return the number of unique clicks in a click dict."""
+    uniq_list = list()
+    for click in clicks:
+        if click["user"] not in uniq_list:
+            uniq_list.append(click["user"])
+    return len(uniq_list)
+
+
+def write_assessment_click_summary(api, assessment_id):
+    """Output a click summary report to JSON, console, and a text file."""
+    assessment_click_summary = dict()
+    campaign_click_stats = dict()
+    campaigns = list()
+    campaign_ids = get_campaign_ids(api, assessment_id)
+    num_campaigns = len(campaign_ids)
+    summary_json = assessment_id + "_click_summary.json"
+    summary_outfile_name = assessment_id + "_click_summary.txt"
+    summary_outfile = open(summary_outfile_name, "w")
+    double_print(summary_outfile, "-" * 50)
+    double_print(summary_outfile, "Number of campaigns: %i" % num_campaigns)
+    assessment_click_summary["num_campaigns"] = num_campaigns
+
+    for campaign_id in campaign_ids:
+        clicks = get_click_data(api, campaign_id)
+        campaign_click_stats["campaign_id"] = campaign_id
+        campaign_click_stats["total_emails_sent"] = api.campaigns.summary(
+            campaign_id=campaign_id
+        ).stats.sent
+        campaign_click_stats["unique_user_count"] = find_unique_target_clicks_count(
+            clicks
+        )
+        if campaign_click_stats["total_emails_sent"] > 0:
+            campaign_click_stats["click_rate"] = float(
+                campaign_click_stats["unique_user_count"]
+            ) / float(campaign_click_stats["total_emails_sent"])
+        else:
+            campaign_click_stats["click_rate"] = 0
```

Comment:
Ensure `campaign_click_stats["click_rate"]` is always a `float`.
```suggestion
            campaign_click_stats["click_rate"] = 0.0
```

Reply:
See 0a3f4bc0587431f616a0d35548f9fc80da72a16b for changes to this line

Reply:
Looks good, though in the future, it's best practice to put a fix like this in its own commit, rather than hide it within a larger commit that addresses other issues.  Better yet, in this case, it would have been simplest to just use the "Commit suggestion" feature in the GitHub web GUI.